src/index.js

```diff
@@ -0,0 +1,78 @@
+import { PRIVATE_FORMATS, PUBLIC_FORMATS, DEFAULT_PRIVATE_FORMATS, DEFAULT_PUBLIC_FORMATS } from './formats';
+import { validateInputKey, validateFormat, validateDecomposedKey } from './util/validator';
+import { AggregatedError } from './util/errors';
+import { KEY_TYPES } from './util/key-types';
+
+const decomposeKey = (supportedFormats, inputKey, options) => {
+    inputKey = validateInputKey(inputKey);
+
+    if (!Array.isArray(options.format)) {
+        const format = validateFormat(options.format, supportedFormats);
+
+        return supportedFormats[format].decomposeKey(inputKey, options);
+    }
+
+    const formats = options.format.map((format) => validateFormat(format, supportedFormats));
+
+    // Attempt to decompose the keys, until one succeeds
+    // Along the way, we collect the errors for each format
+    const errors = {};
+    let decomposeKey;
+
+    for (const format of formats) {
```

Comment:
Agree, I know what you mean. I will try to improve.

Reply:
Changed to:

> When `options.format` is an array, this function will attempt to decompose the key for the specified formats, in order and one by one. It will succeed if the key is using one of the formats or fail if it's using another format, throwing an AggregatedError containing a `errors` property with the errors indexed by format.